staging/src/k8s.io/kubectl/pkg/cmd/wait/wait_test.go

```diff
@@ -1717,3 +1717,42 @@ func TestConditionFuncFor(t *testing.T) {
 		})
 	}
 }
+
+// Test condition that ensures that wait for create continues polling when no resources are found
+func TestWaitForCreateSimple(t *testing.T) {
```

Comment:
I'd expect passing a label selector for non-existed resource and the verification. It would not be easy in unit tests though.
So what about adding in here https://github.com/kubernetes/kubernetes/blob/master/test/cmd/wait.sh
* run creation condition with label selector and expect timeout error and not expect resource not found
* create the resource in background
* run creation condition with label selector and expect success
WDYT?

Reply:
Yes, that's much better. I didn't know there was a test sub-directory 😄 
Should I remove the unit test then?

Reply:
Current unit test does not cover actual functionality and I think we can remove it.

Reply:
Got it.

Reply:
Added in https://github.com/kubernetes/kubernetes/pull/128662/commits/724a470a5fc3b8101641476c53310b5bc95a9bd4